### PR TITLE
add MySessionsRepositoryTest

### PR DESCRIPTION
**Pull request by lVlA**
## Overview (Required)
- test MySessionsRepository

### CODE REVIEW

File: app/src/main/java/io/github/droidkaigi/confsched2017/repository/sessions/MySessionsLocalDataSource.java [hunk 1]
```diff
@@ -16,7 +16,7 @@
 import io.reactivex.android.schedulers.AndroidSchedulers;
 import io.reactivex.schedulers.Schedulers;
 
-public final class MySessionsLocalDataSource implements MySessionsDataSource {
```
Comment (lVlA) [inline app/src/main/java/io/github/droidkaigi/confsched2017/repository/sessions/MySessionsLocalDataSource.java]: to mock

File: app/src/test/java/io/github/droidkaigi/confsched2017/repository/sessions/MySessionsRepositoryTest.kt [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+            localDataSource.verify(Mockito.times(1)).findAll()
+        }
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun save() {
+        val session = newDummySession(1)
+        localDataSource.save(session).invoked.thenReturn(Completable.complete())
+        with(TestObserver<Any>()) {
```
Comment (lVlA) [inline app/src/test/java/io/github/droidkaigi/confsched2017/repository/sessions/MySessionsRepositoryTest.kt]: Is this `<Any>` correct?

File: app/src/test/java/io/github/droidkaigi/confsched2017/repository/sessions/MySessionsRepositoryTest.kt [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+        val expected = Array(10) { i -> newDummyMySession(i) }.toList()
+        localDataSource.findAll().invoked.thenReturn(Single.just(expected))
+
+        with(TestObserver<List<MySession>>()) {
+            repository.findAll().subscribe(this)
+            localDataSource.verify(Mockito.times(1)).findAll()
+            assertNoErrors()
+            assertResult(expected)
+            assertComplete()
+        }
```
Comment (kgmyshin) [inline app/src/test/java/io/github/droidkaigi/confsched2017/repository/sessions/MySessionsRepositoryTest.kt]: [imo]
I think rxjava2-test usually use `test()` method, as following.

```
repository.findAll().test().run {
  assertNoErrors()
  assertResult(expected)
  assertComplete()
}
```

other example:  https://github.com/ReactiveX/RxJava/blob/d3455d0c9d57d522c31b5c25af83e8f2b8df12b6/src/test/java/io/reactivex/flowable/FlowableFuseableTest.java

Comment (lVlA) [inline app/src/test/java/io/github/droidkaigi/confsched2017/repository/sessions/MySessionsRepositoryTest.kt]: thanks!
i think it is better😀

Comment (lVlA) [inline app/src/test/java/io/github/droidkaigi/confsched2017/repository/sessions/MySessionsRepositoryTest.kt]: done.
this method is useful 🤣

Comment (konifar) [inline app/src/test/java/io/github/droidkaigi/confsched2017/repository/sessions/MySessionsRepositoryTest.kt]: @kgmyshin Thanks! Noted 📝

**Comment by konifar** [comment]
LGTM! Thanks! 👍

**Pull request merged**